pydm/tests/widgets/test_base.py

```diff
@@ -0,0 +1,540 @@
+    (PyDMWidget.ALARM_CONTENT, PyDMWidget.ALARM_MINOR, {"color": "#EBEB00"}),
+    (PyDMWidget.ALARM_CONTENT, PyDMWidget.ALARM_MAJOR, {"color": "#EB0000"}),
+    (PyDMWidget.ALARM_CONTENT, PyDMWidget.ALARM_INVALID, {"color": "#EB00EB"}),
+    (PyDMWidget.ALARM_CONTENT, PyDMWidget.ALARM_DISCONNECTED, {"color": "#EBEBEB"}),
+
+    (PyDMWidget.ALARM_CONTENT | PyDMWidget.ALARM_BORDER, PyDMWidget.ALARM_NONE,
+     {"color": "black", "border": "2px solid transparent"}),
+    (PyDMWidget.ALARM_CONTENT | PyDMWidget.ALARM_BORDER, PyDMWidget.ALARM_MINOR,
+     {"color": "#EBEB00", "border": "2px solid #EBEB00"}),
+    (PyDMWidget.ALARM_CONTENT | PyDMWidget.ALARM_BORDER, PyDMWidget.ALARM_MAJOR,
+     {"color": "#EB0000", "border": "2px solid #EB0000"}),
+    (PyDMWidget.ALARM_CONTENT | PyDMWidget.ALARM_BORDER, PyDMWidget.ALARM_INVALID,
+     {"color": "#EB00EB", "border": "2px solid #EB00EB"}),
+    (PyDMWidget.ALARM_CONTENT | PyDMWidget.ALARM_BORDER, PyDMWidget.ALARM_DISCONNECTED,
+     {"color": "#EBEBEB", "border": "2px solid #EBEBEB"}),
+])
+def test_qcolor_for_alarm(qtbot, alarm_type, alarm, expected_qcolor):
+    """
+    Test the QColor object generated for the widget based on an alarm context.
+
+    Expectations:
+    The QColor object will be correctly generated as expected.
+
+    Parameters
+    ----------
+    qtbot : fixture
+        Window for widget testing
+    alarm_type : int
+        The alarm type, e.g. AlARM_CONTENT, ALARM_INDICATOR, ALARM_BORDER, ALARM_CONTENT | ALARM_BORDER, etc.
+    alarm : int
+        The alarm severity, e.g. ALARM_NONE, ALARM_MINOR, ALARM_MAJOR, ALARM_INVALID, ALARM_DISCONNECTED
+    expected_qcolor : QColor
+        The expected QColor to be generated
+    """
+    pydm_label = PyDMLabel()
+    qtbot.addWidget(pydm_label)
+
+    qcolor = pydm_label.qcolor_for_alarm(alarm, alarm_type)
+    assert qcolor == QColor(expected_qcolor["color"])
+
+
+# --------------------
+# NEGATIVE TEST CASES
+# --------------------
+
+@pytest.mark.parametrize("base_class, obj_class_name", [
+    (None, None),
+])
+def test_compose_stylesheet_wrong_base_type(test_alarm_style_sheet_map, base_class, obj_class_name):
+    """
+    Test whether an exception is correctly raised if the wrong base type is provided during style sheet composition.
+
+    Expectations:
+    The correct exception, with the correct error message, is raised
+
+    Parameters
+    ----------
+    test_alarm_style_sheet_map : fixture
+       The widget's style map depending on the alarm event
+    base_class : type
+       The base class of the widget
+    obj_class_name : type
+       The class object of the widget
+   """
+    obj = None
+    if obj_class_name:
+        obj = obj_class_name()
+
+    style = dict(test_alarm_style_sheet_map[PyDMWidget.ALARM_CONTENT][PyDMWidget.ALARM_MAJOR])
+    with pytest.raises(TypeError) as e:
+        compose_stylesheet(style, base_class=base_class, obj=obj)
```

Comment:
I am not sure if `compose_stylesheet` should be raising exceptions.
Let's talk about this case offline. This looks like a bug not like a feature.